### PR TITLE
Option to show nested fields inline, better semantic type handling

### PR DESCRIPTION
**Pull request by gabrielRamaker**
Added an option which allows user to show nested JSON as inline fields.
Added semantic handling for booleans, dates and urls.

### CODE REVIEW

File: JSON-connect/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules
```
Comment (diminishedprime) [inline JSON-connect/.gitignore]: You should be able to remove this file, the parent .gitignore should take care of this.

File: JSON-connect/.clasp.json [hunk 2]
```diff
@@ -1 +1 @@
-{"scriptId":"1y2Cax2-fmQ_Iea9RIG4kc_wTfLevkV3KW4cYvNIxMk9p5qdVpXmnpnU9","rootDir":"src"}
+{"scriptId":"1JVKGb63vmuX7KJUjWKVpOds3rq550FLeL5ZGjBtTb2sAhvzDeJdLRxt7","rootDir":"src"}
```
Comment (diminishedprime) [inline JSON-connect/.clasp.json]: Please revert this change, this is the script id that we use for the deployment.

File: JSON-connect/package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "dsccConnector": {
-    "production": "AKfycbwURHkauguMPzj2gJKaFRYR4z3qqNDDVRngu8YIJHGelfIZBDjbZCQu_TqARz0uaxtb",
-    "latest": "AKfycbzWcrFPGGlkUczJiPXmf7C1q4-8mBez7CwuH7Vinn4"
+    "production": "AKfycbzDZJEtN414a8F_vWQ3z9MHFJKXpXzGz8rq7De5jfPMmJz-k49Dpd4gbW5olVa0BvI",
```
Comment (diminishedprime) [inline JSON-connect/package.json]: Please revert this change, we want to keep the current deployments.

File: JSON-connect/src/main.js [hunk 4]
```diff
@@ -25,12 +25,11 @@
  * @param {String} message The exception message
  */
 function sendUserError(message) {
+  console.log(message);
```
Comment (diminishedprime) [inline JSON-connect/src/main.js]: Please remove any excess calls to console.log.

File: JSON-connect/src/main.js [hunk 5]
```diff
@@ -159,51 +177,144 @@ function fetchData(url, cache) {
...
+  try {
+    var content = cache ? getCachedData(url) : fetchJSON(url);
+  } catch (e) {
+    sendUserError(
+      'Unable to download or cache "' +
+        url +
+        '". Error: \n' +
+        e +
+        '\n' +
+        e.stack
```
Comment (diminishedprime) [inline JSON-connect/src/main.js]: We don't want to show stacktraces to regular users. Try to stick to an error message that makes sense to everyone.

File: JSON-connect/src/main.js [hunk 6]
```diff
@@ -159,51 +177,144 @@ function fetchData(url, cache) {
...
  * @param   {Object}  content The content object
  * @return  {Object}           An object with the connector configuration
  */
-
 function getFields(request, content) {
   var cc = DataStudioApp.createCommunityConnector();
   var fields = cc.getFields();
   var types = cc.FieldType;
   var aggregations = cc.AggregationType;
+  var isInline = request.configParams.nestedData == 'inline';
```
Comment (diminishedprime) [inline JSON-connect/src/main.js]: use `===` for equality comparisons.

File: JSON-connect/src/main.js [hunk 7]
```diff
@@ -213,15 +324,37 @@ function getSchema(request) {
...
+ *  Converts date strings to YYYYMMDDHH:mm:ss
+ *
+ * @param   {String} val  Date string
+ * @returns {String}      Converted date string
+ */
+function convertDate(val) {
+  var date = new Date(val);
+  return (
+    date.getUTCFullYear() +
+    ('0' + (date.getUTCMonth() + 1)).slice(-2) +
```
Comment (diminishedprime) [inline JSON-connect/src/main.js]: Won't this always add a starting 0, even if the month is November or December? Same question for the month & day

File: JSON-connect/.gitignore (hunk 1 above)
Comment (gabrielRamaker) [reply JSON-connect/.gitignore]: Would you mind if kept it? The files are in a different repo too and I'd prefer not to push the node_modules to that repo.

File: JSON-connect/src/main.js (hunk 7 above)
Comment (gabrielRamaker) [reply JSON-connect/src/main.js]: It slices the last two characters, so 012 becomes 12.

File: JSON-connect/.clasp.json (hunk 2 above)
Comment (gabrielRamaker) [reply JSON-connect/.clasp.json]: done

File: JSON-connect/package.json (hunk 3 above)
Comment (gabrielRamaker) [reply JSON-connect/package.json]: done

File: JSON-connect/src/main.js (hunk 5 above)
Comment (gabrielRamaker) [reply JSON-connect/src/main.js]: makes sense, done.

File: JSON-connect/src/main.js (hunk 6 above)
Comment (gabrielRamaker) [reply JSON-connect/src/main.js]: done

File: JSON-connect/src/main.js (hunk 4 above)
Comment (gabrielRamaker) [reply JSON-connect/src/main.js]: done

**Comment by gabrielRamaker** [comment]
> This is a very cool addition. Thanks for the PR. Just left a few comments that need to be addressed before these changes can be merged in.

Thanks! I should credit @tdurieux for most of the job.

File: JSON-connect/src/main.js (hunk 7 above)
Comment (diminishedprime) [reply JSON-connect/src/main.js]: Oh, duh. That's really handy!

File: JSON-connect/.gitignore (hunk 1 above)
Comment (diminishedprime) [reply JSON-connect/.gitignore]: could you do that via a global .gitignore instead? It's not a huge deal, but I'm trying to keep this repo clean and focused if possible. `node_modules` should already be ignored by [this line], right?

[this line]: https://github.com/googledatastudio/community-connectors/blob/master/.gitignore#L44

File: JSON-connect/package-lock.json [hunk 8]
```diff
@@ -0,0 +1,7360 @@
+{
```
Comment (diminishedprime) [inline JSON-connect/package-lock.json]: Could you remove this? we're using yarn for the other repos so we should have yarn.lock's instead of package-lock.json.

Comment (diminishedprime) [reply JSON-connect/package-lock.json]: Sorry I missed this on the first look-over.

File: JSON-connect/.gitignore (hunk 1 above)
Comment (gabrielRamaker) [reply JSON-connect/.gitignore]: I'm committing to the other repo from within the JSON-connect folder, so the community-connectors/.gitignore is... ignored. Didn't think of the global .gitignore file, thanks for the tip!

File: JSON-connect/package-lock.json (hunk 8 above)
Comment (gabrielRamaker) [reply JSON-connect/package-lock.json]: done
